### PR TITLE
[android-auto] Navigation intents support

### PR DESCRIPTION
**Pull request by AndrewShkrob**
>An [Intent](https://developer.android.com/reference/android/content/Intent.html) to navigate.
>
>The action must be [ACTION_NAVIGATE](https://developer.android.com/reference/androidx/car/app/CarContext#ACTION_NAVIGATE()).
>The data URI scheme must be either a latitude,longitude pair, or a + separated string query as follows:
>1) "geo:12.345,14.8767" for a latitude, longitude pair.
>2) "geo:0,0?q=123+Main+St,+Seattle,+WA+98101" for an address.
>3) "geo:0,0?q=a+place+name" for a place to search for.

Currently, the core functionality does not include support for queries 2 and 3. However, the logic for handling these queries has been already implemented and tested to the best extent possible.

I encountered an issue where I couldn't send an intent via ADB. Although the intent was successfully dispatched, it wasn't processed as expected.
`./adb shell am startservice -a androidx.car.app.action.NAVIGATE -n app.organicmaps.debug/app.organicmaps.car.NavigationCarAppService -d "geo:0,0?q=50.06370664014736,20.007851471299205"`

You can test it with another AA application. F.e. `EVMap`

### CODE REVIEW

**Review by Jean-BaptisteC — APPROVED** [review]
✅ Pixel 6 - Android 13 with EVMap
Start routing doesn't work, but I guess is not yet implement :)

**Comment by AndrewShkrob** [comment]
> ✅ Pixel 6 - Android 13 with EVMap
> Start routing doesn't work, but I guess is not yet implement :)

What do you mean?
After "Navigate" button click in EVMap, Place Screen should appear where you can click "Route to" button

**Comment by Jean-BaptisteC** [comment]
Route to button doesn't work for me (nothing in logcat)

**Comment by AndrewShkrob** [comment]
> Route to button doesn't work for me (nothing in logcat)

I guess, it's location problem. Like when OM says `Navigation is only available from your current location.`

Try to click on location button and then `Route to`

**Comment by Jean-BaptisteC** [comment]
Yes you right

**Comment by AndrewShkrob** [comment]
That's, probably, because of recent `LocationHelper` refactoring in master. I rebased branch on it and broke some logic. Permissions Request Screen fixes it

**Comment by biodranik** [comment]
@AndrewShkrob are normal coordinates instead of 0,0 also supported in AA and the main app? Is it feasible to fix?

**Comment by AndrewShkrob** [comment]
AA supports only 3 URI schemes from the PR description.
As I already mentioned, `"geo:12.345,14.8767"` works fine right now.

> AA supports only 3 URI schemes from the PR description.

It means that other AA apps will send navigation intents to our AA app using only these 3 URI schemes.
That doesn't mean we cannot add support of all other schemes from https://omaps.app/test in the future.
Example workflow:
1. AA is running
2. Open link on the phone (f.e. om://o4B4pYZsRs)
3. Show data on the AA screen

But I won't implement it right now. Let's leave it for the next iteration.

Both 2nd and 3rd queries are not supported in our core:
> 2. "geo:0,0?q=123+Main+St,+Seattle,+WA+98101" for an address.
> 3. "geo:0,0?q=a+place+name" for a place to search for.

File: android/src/app/organicmaps/intent/Factory.java [hunk 1]
```diff
@@ -44,6 +43,21 @@ public static class GeoIntentProcessor implements IntentProcessor
...
     public MapTask process(@NonNull Intent intent)
+    {
+      final String uri = processIntent(intent);
+      if (uri == null)
+        return null;
+      return new OpenUrlTask(uri);
+    }
+
+    @Nullable
+    public String processCarIntent(@NonNull Intent intent)
```
Comment (rtsisyk) [inline android/src/app/organicmaps/intent/Factory.java:54]: Just kill this useless wrapper.

**Comment by rtsisyk** [comment]
> Both 2nd and 3rd queries are not supported in our core:
> 
> > 2. "geo:0,0?q=123+Main+St,+Seattle,+WA+98101" for an address.
> > 3. "geo:0,0?q=a+place+name" for a place to search for.

This issue is tracked in https://github.com/organicmaps/organicmaps/issues/320.

**Comment by AndrewShkrob** [comment]
@rtsisyk PTAL

**Review by rtsisyk — APPROVED** [review]

**Pull request merged**